src/senaite/sync/browser/views.py

```diff
@@ -355,6 +375,31 @@ def translate_path(self, path):
         remote_portal_id = path.split("/")[1]
         return path.replace(remote_portal_id, portal_id)
 
+    def fetch_registry_records(self, domain, keywords=None):
+        """Fetch configuration registry records of interest (those associated
+        to the keywords passed) from source instance
+        """
+        logger.info("*** FETCH REGISTRY RECORDS {} ***".format(domain))
+        storage = self.get_storage(domain=domain)
+        registry_store = storage["registry"]
+        retrieved_records = {}
+        # If keyword is None retrieve the whole registry
+        if keywords is None:
+            retrieved_records["all"] = self.get_registry_records_by_keyword()
```

Comment:
Refactor like this to make it more readable (maybe) 🤔

```python
if keys is None:
    keys = []
for key in keys:
    records[key] = self.get_registry_records_by_key(key)
    [...]
else:
    records["all"] = self.get_registry_records_by_key(key)
```

I would also refactor `self.get_registry_records_by_key(key)` to return the contents of `items` immediately

Reply:
@ramonski I tried to refactor it as specified but I was facing a problem where the `else` clause was being always executed independently of the value of `keys`. As a results I was getting not only the registry records associated to bika or senaite but all of them.